src/midi/midi.cpp

```diff
@@ -149,6 +149,21 @@ class MidiState {
 		note_on_tracker.fill(false);
 	}
 
+	void Track(const MidiMessage& msg)
+	{
+		const auto status  = get_midi_status(msg.status());
+		const auto channel = get_midi_channel(msg.status());
+
+		if (status == MidiStatus::NoteOn) {
+			const auto note = msg.data1();
+			SetNoteActive(channel, note, true);
+
+		} else if (status == MidiStatus::NoteOff) {
+			const auto note = msg.data1();
+			SetNoteActive(channel, note, false);
+		}
```

Comment:
The `Track(msg)` function name sounds like we're just keeping track of the message, like a back-end registry or something - but in the body, we're calling `SetNoteActive` and actually turning it on or off.

Not sure if the function name needs to change  maybe `TrackAndActivate`? or something that implies that this isn't just a weight-in station before the message lorry carries on :-)

I might also be confusing "Track" here.. perhaps "Track" is a more like an Audio Track - a sequence of messages?

Reply:
> The Track(msg) function name sounds like we're just keeping track of the message, like a back-end registry or something - but in the body, we're calling SetNoteActive and actually turning it on or off.

Nope, we're only tracking. Yes, this was also outputting MIDI messages in its previous incarnation, but I've de-conflated that stuff; not it's only tracking the status, as the class name implies.

`SetNoteActive` and `SetChannelVolume` only set the internal status.

> I might also be confusing "Track" here.. perhaps "Track" is a more like an Audio Track - a sequence of messages?

Good point, renamed to `UpdateState`.

Reply:
Also added a class comment.